packages/hub/node-tests/tasks/send-notifications-test.ts

```diff
@@ -0,0 +1,125 @@
+import { buildConditions } from '../../utils/queries';
+
+// https://github.com/graphile/worker/blob/e3176eab42ada8f4f3718192bada776c22946583/__tests__/helpers.ts#L135
+export function makeMockJob(taskIdentifier: string): Job {
+  const createdAt = new Date(Date.now() - 12345678);
+  return {
+    id: String(Math.floor(Math.random() * 4294967296)),
+    queue_name: null,
+    task_identifier: taskIdentifier,
+    payload: {},
+    priority: 0,
+    run_at: new Date(Date.now() - Math.random() * 2000),
+    attempts: 0,
+    max_attempts: 25,
+    last_error: null,
+    created_at: createdAt,
+    updated_at: createdAt,
+    locked_at: null,
+    locked_by: null,
+    revision: 0,
+    key: null,
+    flags: null,
+  };
+}
+
+let helpers = makeJobHelpers({}, makeMockJob('send-notifications'), {
+  withPgClient: () => {
+    throw new Error('withPgClient is not implemented in test');
+  },
+});
+
+const messageID = 'firebase-message-id';
+let createPushNotification: (prefix: string) => PushNotificationData = (prefix = '') => ({
+  transactionHash: `${prefix}transaction-hash`,
+  // keep owner-address and pushClientId the same since this is a likely occurence in prod
+  // when we send several notifications for different transactions
+  ownerAddress: `owner-address`,
+  pushClientId: `push-client-id`,
+  notificationTitle: `${prefix}notification-title`,
+  notificationBody: `${prefix}notification-body`,
+  notificationData: [],
+  notificationType: `${prefix}mock`,
+  network: 'test-network',
+});
+let existingNotification = createPushNotification('existing-');
+let newlyAddedNotification = createPushNotification('newly-added-');
+
+let lastSentData: any;
+let notificationSent = false;
+
+class StubFirebasePushNotifications {
+  async send(data: any) {
+    lastSentData = data;
+    notificationSent = true;
+    return messageID;
+  }
+}
+
+describe('SendNotificationsTask', function () {
+  let subject: SendNotifications;
+  let db: Client;
+  let { getContainer } = setupHub(this);
+
+  this.beforeEach(async function () {
+    let dbManager = await getContainer().lookup('database-manager');
+    db = await dbManager.getClient();
+    await db.query(`DELETE FROM sent_push_notifications`);
+    await db.query(
+      `INSERT INTO sent_push_notifications (transaction_hash, owner_address, push_client_id, notification_type, notification_title, notification_body, notification_data, message_id, network) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)`,
+      [
+        existingNotification.transactionHash,
+        existingNotification.ownerAddress,
+        existingNotification.pushClientId,
+        existingNotification.notificationType,
+        existingNotification.notificationTitle,
+        existingNotification.notificationBody,
+        existingNotification.notificationData,
+        'existing-message-id',
+        existingNotification.network,
+      ]
+    );
+    subject = (await getContainer().lookup('send-notifications')) as SendNotifications;
```

Comment:
Could you use the queries class here?

Reply:
For database interactions? I was wondering about that... using the queries class is more readable, but went with this because it felt like the database manager would be a lot less likely to have additional logic introduced in the future/ is a lot more stable.

Reply:
I would encourage to use queries services as much as possible, I think it will be easier to refactor later on if we decide to introduce an ORM.

Reply:
Database interactions (besides wiping the sent-push-notifications db each time) are now done with the query service